### PR TITLE
new distribution with pledge and allocation info

### PR DESCRIPTION
**Pull request by srkirkland**

### CODE REVIEW

File: GiftModels/GiftDetails.cs [hunk 1]
```diff
@@ -30,6 +30,13 @@ public GiftDetails()
         [Required]
         [DataAnnotationsHelper.MinListLength(1)]
         public IList<AllocationContainer> Allocations { get; set; }
+        
+        public AllocationDetail PrimaryAllocation
```
Comment (jpknoll) [inline GiftModels/GiftDetails.cs]: @srkirkland I'd be happier with just "Allocation".

**Comment by jpknoll** [comment]
@srkirkland Soon as this goes live, we'll need to write some migration code to the new distributions property.

I like the idea of dropping the Allocations array eventually too.

File: GiftModels/GiftDetails.cs (hunk 1 above)
Comment (srkirkland) [inline GiftModels/GiftDetails.cs]: @jpknoll works for me

**Comment by srkirkland** [comment]
@jpknoll we don't really need to except for those gifts in a pre-batched stage though right?  So shouldn't matter in anything out there currently.

**Comment by jpknoll** [comment]
@srkirkland I'd be concerned about gifts in the history not serializing. We might want that data for reporting.

**Comment by srkirkland** [comment]
@jpknoll before getting rid of the allocations array for sure, but until then not sure why we'd need to start populating distributions array

**Comment by jpknoll** [comment]
@srkirkland Add frequency and amount to the pledge.

**Comment by jpknoll** [comment]
:shipit:

**Pull request merged**